### PR TITLE
refactor application letter factory

### PR DESCRIPTION
**Pull request by aloeser**
we now use traits instead of inheritance

### CODE REVIEW

**Comment by aloeser** [comment]
One thing I'm not sure about is whether to put the traits into or below the factories, like

```
factory :application_letter do

    trait :accepted do
          status :acccepted
    end
end

trait :rejected do
    status :rejected
end
```

Any comments or ideas?

**Comment by aloeser** [comment]
Nevermind, 1min debugging with `rails console` helped. Putting them below factories makes them globally available, so that you can run `FactoryGirl.build(:event, :accepted)` which doesnt make sense.
I'll change it soon

File: spec/controllers/email_controller_spec.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
     context "with valid accepted applications" do
       before :each do
-        @application = FactoryGirl.create(:application_letter_accepted, event: @event, user: FactoryGirl.build(:user))
+        @application = FactoryGirl.create(:application_letter, :accepted, event: @event, user: FactoryGirl.build(:user))
```
Comment (querenker) [inline spec/controllers/email_controller_spec.rb]: Is the user association in this create call even necessary? I think [this association definition](https://github.com/hpi-schuelerklub/workshop-portal/blob/35d5b8dd50ac72141dd33b35200a3682b5712588/spec/factories/application_letters.rb#L22) should be enough?

File: spec/controllers/email_controller_spec.rb [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
     context "with valid rejected applications" do
       before :each do
-        @application = FactoryGirl.create(:application_letter_rejected, event: @event, user: FactoryGirl.build(:user))
+        @application = FactoryGirl.create(:application_letter, :rejected, event: @event, user: FactoryGirl.build(:user))
```
Comment (querenker) [inline spec/controllers/email_controller_spec.rb]: User association again

**Review by querenker — CHANGES_REQUESTED** [review]
The traits are very nice 👍

**Review by querenker — APPROVED** [review]

**Review by bjrne — APPROVED** [review]
Good 🏃‍♀️

**Pull request merged**